rules/utils/element.js

```diff
@@ -72,6 +85,45 @@ function formatTypes(types, exclusive = false) {
 
 module.exports.formatTypes = formatTypes;
 
+module.exports.hasDuplicatedPropertyValues = function(node, propertiesName, propertyName, parentNode = null) {
+  const properties = node.get(propertiesName);
+
+  const propertyValues = properties.map(property => property.get(propertyName));
+
+  const duplicates = propertyValues.reduce((duplicates, propertyValue, index) => {
+    if (propertyValues.indexOf(propertyValue) !== index && !duplicates.includes(propertyValue)) {
+      return [
+        ...duplicates,
+        propertyValue
+      ];
+    }
+
+    return duplicates;
+  }, []);
+
+  if (duplicates.length) {
+    return duplicates.map(duplicate => {
+      const duplicateProperties = properties.filter(property => property.get(propertyName) === duplicate);
+
+      return {
+        message: `Properties of type <${ duplicateProperties[ 0 ].$type }> have property <${ propertyName }> with duplicate value of <${ duplicate }>`,
+        path: null,
+        error: {
+          type: ERROR_TYPES.PROPERTY_VALUE_DUPLICATED,
+          node,
+          parentNode: parentNode == node ? null : parentNode,
+          duplicatedProperty: propertyName,
+          duplicatedPropertyValue: duplicate,
+          properties: duplicateProperties,
```

Comment:
Took me a bit why we are detecting the duplicates first and then iterating again, but having all duplicated properties to report it via `properties` made it clear to me 👍

Reply:
Yes, so that information is necessary so we can display the errors in the properties panel.